content/articles/building-a-simple-translation-app-using-python-for-beginners/index.md

```diff
@@ -0,0 +1,171 @@
+
+```
+
+Amazing right? The translation pipeline is translating our text from English to French. 
+
+Lastly, we are going to create a function to wrap up all these and display it on a beautiful user interface. We'll be leveraging Gradio for this task.
+
+### Building the web app for interaction using Gradio
+This web app allows a user to pass through some text and have it translated.
+
+```python
+    def translate_transformers(from_text):
+    translated_results = translation_pipeline(from_text)
+    return translated_results[0]['translation_text']
+```
+
+We have created a function called `translate_transformers()`. We then pass through the text to our translate transformer pipeline. We are storing the results of the translation in a variable called `translated_results`. The function then grabs the first value in our list [0] and returns the translated text.
+
+Now, all that's left to do is wrap this up inside the Gradio function.
+
+We create the gradio interface by writing following pieces of code:
+
+```python
+interface = gr.Interface(fn=translate_transformers, 
+                        inputs=gr.inputs.Textbox(lines=2, placeholder='Text to translate'),
+                        outputs='text')
+```
+It's amazing how quickly you can create this interface, with only a single line of code. We've passed in a few arguments. `fn` is where we pass in our function which is `translate_transformers`. We pass in an `input` type of Textbox with 2 lines and a placeholder. You can change the number of lines and placeholders in your project without any issues. The last argument states the output type. For our case, it's `text`. 
+
+Let's go ahead and launch it. We use the `launch()` method to achieve this. 
+
+```python
+interface.launch()
+```
+This command will run our gradio app.
+
+![Gradio app](/engineering-education/building-a-simple-translation-app-using-python-for-beginners/gradio-app.png)
+
+*[Image Source: Colab](https://colab.research.google.com/drive/1MMgN89Xn7fz5zVh1XCuLd66ZBIoZ70DZ#scrollTo=62a4vvW_dLsx)*
+
+You can also run the gradio app using the external link generated after issuing the `launch()` instruction. For my case, this is the link that is generated `https://18497.gradio.app`.
+
+![Gradio translation](/engineering-education/building-a-simple-translation-app-using-python-for-beginners/gradio-translation.png)
+
+*[Image Source: Colab](https://colab.research.google.com/drive/1MMgN89Xn7fz5zVh1XCuLd66ZBIoZ70DZ#scrollTo=62a4vvW_dLsx)*
+
+Our code works successfully by using gradio to translate.
+
+### Wrapping up
+This tutorial gives us an idea of how one can build a simple translation app. We installed our dependencies, loaded our translation model, and finally used gradio to build our user interface. This is a good example of a project that you can add to your resume or portfolio. It will show the employer of your Machine Learning capabilities. You can play around with the code by trying to translate languages other than french.
```

Comment:
"Machine Learning" can be replaced with "machine learning", and capitalize the word "french"